v2/pkg/engine/datasource/graphql_datasource/graphql_tws_handler.go

```diff
@@ -259,6 +273,32 @@ func (h *gqlTWSConnectionHandler) handleMessageTypePing() {
 	}
 }
 
+func (h *gqlTWSConnectionHandler) handleMessageTypePong() {
+	h.lastPongReceivedAt = time.Now()
+}
+
+func (h *gqlTWSConnectionHandler) Ping() {
+
+	// Check if we have received a pong message in the pongWaitTimeout
+	// If not we assume the connection is dead and close it
+	if !h.lastPingSentAt.IsZero() {
+		if h.lastPongReceivedAt.Sub(h.lastPingSentAt) > pongWaitTimeout {
```

Comment:
Can you make this a function that is easy to understand by name?
e.g. "hasReceivedPongBeforeTimeout" or similar

Reply:
Done